src/ring_theory/fractional_ideal.lean

```diff
@@ -336,6 +337,72 @@ instance comm_semiring : comm_semiring (fractional_ideal f) :=
   ..fractional_ideal.add_comm_monoid,
   ..fractional_ideal.comm_monoid }
 
+variables {P' : Type*} [comm_ring P'] {f' : localization_map S P'}
+variables {P'' : Type*} [comm_ring P''] {f'' : localization_map S P''}
+
+lemma fractional_map (g : f.codomain →ₐ[R] f'.codomain) (I : fractional_ideal f) :
+  is_fractional f' (submodule.map g.to_linear_map I.1) :=
+begin
+  rcases I with ⟨I, a, a_nonzero, hI⟩,
+  use [a, a_nonzero],
+  intros b hb,
+  obtain ⟨b', b'_mem, hb'⟩ := submodule.mem_map.mp hb,
+  obtain ⟨x, hx⟩ := hI b' b'_mem,
+  use x,
+  erw [←g.commutes, hx, g.map_smul, hb'],
+  refl
+end
+
+/-- `I.map g` is the pushforward of the fractional ideal `I` along the algebra morphism `g` -/
+def map (g : f.codomain →ₐ[R] f'.codomain) :
+  fractional_ideal f → fractional_ideal f' :=
+λ I, ⟨submodule.map g.to_linear_map I.1, fractional_map g I⟩
+
+@[simp] lemma val_map (g : f.codomain →ₐ[R] f'.codomain) (I : fractional_ideal f) :
+(map g I).val = submodule.map g.to_linear_map I.val := rfl
```

Comment:
Shouldn't this use `coe`?

Reply:
Which coercion do you mean, because `algebra_map.to_linear_map` and `fractional_ideal.val` don't have coercions?

Reply:
I see. Should `fractional_ideal.val` become a coercion?

Reply:
I added the instance. Do you feel it improves readability? It requires a lot more explicit type annotations.

Reply:
I agree, but I think the type ascriptions are more readable then `.1`. So I personally like this change.